app/src/main/kotlin/com/saltedge/authenticator/features/launcher/LauncherViewModel.kt

```diff
@@ -22,20 +22,39 @@ package com.saltedge.authenticator.features.launcher
 
 import android.app.Activity
 import android.content.Context
+import androidx.lifecycle.*
 import com.saltedge.authenticator.features.main.MainActivity
 import com.saltedge.authenticator.features.onboarding.OnboardingSetupActivity
+import com.saltedge.authenticator.features.settings.mvvm.about.ViewModelEvent
+import com.saltedge.authenticator.model.realm.RealmManagerAbs
 import com.saltedge.authenticator.model.repository.PreferenceRepositoryAbs
+import com.saltedge.authenticator.tool.AppTools
 import com.saltedge.authenticator.tool.secure.PasscodeToolsAbs
-import javax.inject.Inject
 
-class LauncherPresenter @Inject constructor(
-    private val appContext: Context,
-    private val preferenceRepository: PreferenceRepositoryAbs,
-    private val passcodeTools: PasscodeToolsAbs
-) {
+class LauncherViewModel(
+    val appContext: Context,
+    val preferenceRepository: PreferenceRepositoryAbs,
+    val passcodeTools: PasscodeToolsAbs,
+    val realmManager: RealmManagerAbs
+) : ViewModel(), LifecycleObserver {
 
-    fun setupApplication() {
-        if (shouldSetupApplication()) passcodeTools.replacePasscodeKey(appContext)
+    var errorOccurred = MutableLiveData<ViewModelEvent<Boolean>>()
+        private set
+
+    var noErrorOccurred = MutableLiveData<ViewModelEvent<Boolean>>()
+        private set
+
+    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
+    fun onLifeCycleResume() {
+        if (!AppTools.isTestsSuite(appContext)) realmManager.initRealm(context = appContext)
+        if (realmManager.errorOccurred) {
+            noErrorOccurred.value = null
+            errorOccurred.postValue(ViewModelEvent(true))
```

Comment:
I see that you are using just two states of value: `true` and `null`. 
I think you should use `Unit` type instead of `Boolean` type.

Reply:
u mean 
`errorOccurred.postValue(ViewModelEvent(true))`, but then nothing will happen, I need to post the `value`.

Reply:
You post the ViewModelEvent, not it value